src/main/java/fi/otavanopisto/kuntaapi/server/controllers/EmergencyController.java

```diff
@@ -3,48 +3,81 @@
+import fi.otavanopisto.kuntaapi.server.id.IdController;
 import fi.otavanopisto.kuntaapi.server.id.OrganizationId;
+import fi.otavanopisto.kuntaapi.server.index.SearchResult;
+import fi.otavanopisto.kuntaapi.server.index.search.EmergencySearcher;
 import fi.otavanopisto.kuntaapi.server.integrations.EmergencyProvider;
 import fi.otavanopisto.kuntaapi.server.integrations.EmergencySortBy;
+import fi.otavanopisto.kuntaapi.server.integrations.KuntaApiConsts;
 import fi.otavanopisto.kuntaapi.server.integrations.SortDir;
-import fi.otavanopisto.kuntaapi.server.utils.ListUtils;
-import fi.metatavu.kuntaapi.server.rest.model.Emergency;
 
 @ApplicationScoped
 @SuppressWarnings ({"squid:S3306","squid:S00107"})
 public class EmergencyController {
   
   @Inject
-  private EntityController entityController;
+  private Logger logger;
+
+  @Inject
+  private EmergencySearcher emergencySearcher;
+  
+  @Inject
+  private IdController idController;
   
   @Inject
   private Instance<EmergencyProvider> emergencyProviders;
   
-  public List<Emergency> listEmergencies(OrganizationId organizationId, String location, OffsetDateTime before, OffsetDateTime after, EmergencySortBy sortBy, SortDir sortDir, Integer firstResult, Integer maxResults) {
-    List<Emergency> result = new ArrayList<>();
-   
-    for (EmergencyProvider emergencyProvider : getEmergencyProviders()) {
-      result.addAll(emergencyProvider.listOrganizationEmergencies(organizationId, location, before, after));
+  /**
+   * Search emergencies. 
+   * 
+   * @param organizationId organization id
+   * @param location emergency location
+   * @param before filter by time before. Optional
+   * @param after filter by time after. Optional
+   * @param sortBy sort by
+   * @param sortDir sort direction
+   * @param firstResult first result index
+   * @param maxResults max results
+   * @return result
+   */
+  public SearchResult<Emergency> searchEmergencies(OrganizationId organizationId, String location, OffsetDateTime before, OffsetDateTime after, EmergencySortBy sortBy, SortDir sortDir, Integer firstResult, Integer maxResults) {
+    OrganizationId kuntaApiOrganizationId = idController.translateOrganizationId(organizationId, KuntaApiConsts.IDENTIFIER_NAME);
+    if (kuntaApiOrganizationId == null) {
+      logger.severe(() -> String.format("Failed to translate organization %s into Kunta API id", organizationId.toString()));
+      return SearchResult.emptyResult();
     }
     
-    if (sortBy == EmergencySortBy.NATURAL) {
-      result = entityController.sortEntitiesInNaturalOrder(result, sortDir);
-    } else {
-      Collections.sort(result, new EmergencyTimeComparator(sortDir));
+    SearchResult<EmergencyId> searchResult = emergencySearcher.searchEmergencies(kuntaApiOrganizationId.getId(), null, location, before, after, sortBy, sortDir, firstResult, maxResults);
+    if (searchResult != null) {
+      List<Emergency> result = searchResult.getResult().stream().map((emergencyId) -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement and then remove useless return keyword [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1602)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "emergencyId" parameter [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1611)